_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,302 @@
+
+```json
+POST _ingest/pipeline/dot-expander/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "field": "city, state"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "field": "city, state"
+        },
+        "_ingest": {
+          "timestamp": "2023-11-17T23:49:27.597933805Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+**Step 3: Ingest a document.**
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=dot-expander
+{
+  "field": "Denver, CO"
+}
+```
+{% include copy-curl.html %}
+
+**Step 4 (Optional): Retrieve the document.**
+
+To retrieve the document, run the following query:
+
+```json
+GET testindex1/_doc/1
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following response confirms the document was indexed:
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 58,
+  "_seq_no": 57,
+  "_primary_term": 30,
+  "found": true,
+  "_source": {
+    "field": "Denver, CO"
+  }
+}
+```
+
+## Nested fields
+
+The processor consolidates the `user.address.city` and `user.address.state` fields by merging with an existing `address`, `city`, and `state` field nested under `user`. If the field is a scalar value, then it will turn that into an array field. Take for example the following document:
+
+```json
+{
+    "dot_expander": {
+        "field": "user.address.city",
+        "override": true
+    }
+}
+```
+
+With `override:false` (or unset), the dot-pathed top-level field gets appended to the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": ["Denver", "Boulder"],
+            "state": "CO"
+        }
+    }
+}
+```
+
+With `override:true`, the dot-pathed top-level field value overwrites the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": "Boulder",
+            "state": "CO"
+        }
+    }
+}
+```
+
+If the field value is set to a wildcard `*`, the processor expands all top-level dotted field names, for example:  
+
+```json
+{
+    "dot_expander": {
+        "field": "*"
+    }
+}
+```
+
+Take for example the following:
+
+```json
+{
+    "user.address.city": "Denver",
+    "user.address.state": "CO"
+}
+```
+
+The `dot_expander` processor transforms that document into the following:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": "Denver"
+        }
+    },
+    "user": {
+        "address": {
+            "state": "CO"
+        }
+    }
+}
+```
+
+If the field is nested within a structure without dots, use the `path` parameter to navigate the non-dotted structure, for example:
```

Comment:
What do you mean by "navigate"?

Reply:
I would also clarify that while the nested field at the top layer doesn't have dots, the fields nest beneath them, for example `city.one`, do. So `dot_expander` will take the dotted fields nested underneath the non-dotted field and turns them into an array?